### PR TITLE
Unify the result interfaces from get and search in Java client

### PR DESCRIPTION
**Pull request by olcbean**
As `GetField` and `SearchHitField` have the same members, they have been unified into
`DocumentField`.

Addresses #16440

### CODE REVIEW

**Comment by cbuescher** [comment]
@elasticmachine test this please

File: core/src/main/java/org/elasticsearch/search/SearchHitField.java [hunk 1]
```diff
@@ -1,126 +0,0 @@
...
-import org.elasticsearch.index.mapper.MapperService;
-
-import java.io.IOException;
-import java.util.ArrayList;
-import java.util.Iterator;
-import java.util.List;
-import java.util.Objects;
-
-/**
- * A single field name and values part of a {@link SearchHit}.
```
Comment (cbuescher) [inline core/src/main/java/org/elasticsearch/search/SearchHitField.java]: nit: Can something similar to this become part of a new class javadoc for DocumentField?

File: core/src/main/java/org/elasticsearch/common/document/DocumentField.java [hunk 2]
```diff
@@ -36,34 +36,46 @@
 import static org.elasticsearch.common.xcontent.XContentParserUtils.ensureExpectedToken;
 import static org.elasticsearch.common.xcontent.XContentParserUtils.parseStoredFieldsValue;
 
-public class GetField implements Streamable, ToXContent, Iterable<Object> {
+public class DocumentField implements Streamable, ToXContent, Iterable<Object> {
```
Comment (cbuescher) [inline core/src/main/java/org/elasticsearch/common/document/DocumentField.java]: nit: Can we have some javadoc here, maybe a bit longer that the current SearchHitField javadoc that also states that this is used in both SearchHit and GetResult now?

**Review by cbuescher — COMMENTED** [review]
@olcbean great change, I'm suprised how similar those two classes are. I left only one small ask about moving/extending some class javadoc, doesn't have to be long though. I would also wait if @javanna has anything to add, and we need to adapt some internal code to this before this can be merged, but overall LGTM.

**Comment by s1monw** [comment]
@elasticmachine ok to test

**Comment by olcbean** [comment]
@cbuescher, nice catch!
I just added some class javadoc.

**Review by cbuescher — APPROVED** [review]
Thanks, I will merge this in as soon as the final CI build has passed

**Comment by cbuescher** [comment]
@olcbean thanks again, merged to master

**Pull request merged**